mediathread/djangosherd/permissions.py

```diff
@@ -0,0 +1,38 @@
+from django.core.exceptions import PermissionDenied
+from rest_framework.permissions import BasePermission
+
+from mediathread.main.course_details import (
+    all_items_are_visible, cached_course_is_faculty,
+    cached_course_is_member
+)
+
+
+class AssetIsVisible(BasePermission):
+    message = 'You can only update assets visible to you.'
+
+    def has_object_permission(self, request, view, obj):
+        # Am I the owner of this asset?
+        if obj.author == request.user:
+            return True
+
+        if cached_course_is_faculty(obj.course, request.user):
+            return True
+
+        if not all_items_are_visible(request.course):
+            raise PermissionDenied
+
+        return True
+
+
+class AssetInMyCourse(BasePermission):
+    message = 'You can only update assets in your courses.'
+
+    def has_object_permission(self, request, view, obj):
+        # Am I the owner of this asset?
```

Comment:
This first check duplicates `AssetIsVisible`.  I don't think it is necessary here.